### PR TITLE
cvmfs-with-clang - 2nd try

### PR DESCRIPTION
**Pull request by claudiok**
Apparently making repositories public prevents github from updating pull requests. Here's a new one

### CODE REVIEW

File: builders/tools/cdk.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 def install(dir_name,version=None):
     if not os.path.exists(os.path.join(dir_name,'lib','libcdk.a')):
-        version = '5.0-20150928'
+        version = '5.0-20160131'
```
Comment (dsschult) [inline builders/tools/cdk.py]: Can I reject this version change?  I'm guessing you did this when we still downloaded `cdk.tar.gz`, and needed to keep up with the latest version.  I patched that in the meantime, so we can actually use a fixed version number.

Comment (claudiok) [inline builders/tools/cdk.py]: I changed my pull request (which now works again :) )

File: builders/tools/pip.py [hunk 2]
```diff
@@ -28,7 +27,7 @@ def install_pip(dir_name,version=None):
             tmp_dir = tempfile.mkdtemp()
             path = os.path.join(tmp_dir,name)
             wget('https://bootstrap.pypa.io/get-pip.py',path)
-            if subprocess.call([os.path.join(dir_name,'bin','python'),path,'--quiet']):
+            if subprocess.call([os.path.join(dir_name,'bin','python'),path]):
```
Comment (dsschult) [inline builders/tools/pip.py]: Was there a reason `--quiet` was removed?  Just wondering.

Comment (claudiok) [inline builders/tools/pip.py]: Good question. I don't remember. I guess I wanted to see where things failed while building them and removed it for that (and then never added it back). Feel free to add it back in.

File: builders/tools/sqlite.py [hunk 3]
```diff
@@ -29,12 +30,15 @@ def install(dir_name,version=None):
             wget(url,path)
             unpack(path,tmp_dir)
             sqlite_dir = os.path.join(tmp_dir,'sqlite-autoconf-'+str(version))
+
+            mod_env = copy.deepcopy(os.environ)
```
Comment (dsschult) [inline builders/tools/sqlite.py]: Not sure we need deepcopy here.  I think just `dict(os.environ)` works as well.

File: builders/tools/suitesparse.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ def install(dir_name,version=None):
                 f.write('LAPACK = -L'+os.path.join(dir_name,'lib')+' -lopenblas\n')
             finally:
                 f.close()
-            if subprocess.call(['make'],cwd=suitesparse_dir):
+            if subprocess.call(['make','library'],cwd=suitesparse_dir):
```
Comment (dsschult) [inline builders/tools/suitesparse.py]: Can you tell me the difference between `make` and `make library`, just for my knowledge?

File: builders/tools/zmq.py [hunk 5]
```diff
@@ -19,7 +19,7 @@ def install(dir_name,version=None):
             unpack(path,tmp_dir)
             zmq_dir = os.path.join(tmp_dir,'zeromq-'+version)
             if subprocess.call([os.path.join(zmq_dir,'configure'),
-                                '--prefix',dir_name],cwd=zmq_dir):
+                                '--prefix',dir_name,'--without-libsodium'],cwd=zmq_dir):
```
Comment (dsschult) [inline builders/tools/zmq.py]: Does `--without-libsodium` mean zmq can't use SSL?  Or does it fall back to something else?

File: builders/tools/suitesparse.py (hunk 4 above)
Comment (claudiok) [inline builders/tools/suitesparse.py]: `make` also runs tests, `make library` doesn't. I think `make library` ran into a linking problem with `gcc_s` required by gfortran and failed. This was the easiest way to fix it.

File: builders/tools/sqlite.py (hunk 3 above)
Comment (claudiok) [inline builders/tools/sqlite.py]: don't know. `deepcopy` seemed like the safe thing to do

File: builders/tools/zmq.py (hunk 5 above)
Comment (claudiok) [inline builders/tools/zmq.py]: means it cannot use SSL. If we install libsodium first we could use SSL. Do we have a use case for this? I guess we might in the future. But then that should be a feature request.

Comment (claudiok) [inline builders/tools/zmq.py]: It just wouldn't build for me without the flag

**Comment by dsschult** [comment]
I'm wondering what `icecube.opensciencegrid.org/data/.cannotbeempty` is for.

File: builders/tools/zmq.py (hunk 5 above)
Comment (dsschult) [inline builders/tools/zmq.py]: You'd have to ask if anyone really needs it, but I like security and count it as a useful thing to have.

**Comment by claudiok** [comment]
git doesn't support empty directories, so building failed for me without `data/` existing in there

File: builders/tools/zmq.py (hunk 5 above)
Comment (claudiok) [inline builders/tools/zmq.py]: I don't think we have it in any current version we use.

**Comment by dsschult** [comment]
that's just sad.  it has all these other fancy features, but can't support an empty directory.

**Comment by claudiok** [comment]
I assume your local copy just has the `data/` directory, so you never noticed

**Comment by dsschult** [comment]
probably, or at least the `data/` directory exists on cvmfs already.

**Pull request merged**